packages/web/src/common/store/upload/sagas.ts

```diff
@@ -0,0 +1,1037 @@
+import {
+  Collection,
+  CollectionMetadata,
+  ID,
+  Kind,
+  Name
+} from '@audius/common/models'
+import { CollectionValues } from '@audius/common/schemas'
+import {
+  TrackMetadataForUpload,
+  LibraryCategory,
+  ProgressStatus,
+  TrackForUpload,
+  UploadType,
+  accountActions,
+  accountSelectors,
+  cacheActions,
+  cacheUsersSelectors,
+  confirmTransaction,
+  confirmerActions,
+  getContext,
+  reformatCollection,
+  savedPageActions,
+  uploadActions
+} from '@audius/common/store'
+import {
+  actionChannelDispatcher,
+  makeUid,
+  waitForAccount
+} from '@audius/common/utils'
+import { EntityManagerAction } from '@audius/sdk'
+import type { ProgressCB } from '@audius/sdk/dist/services/creatorNode'
+import type { TrackMetadata, UploadTrackMetadata } from '@audius/sdk/dist/utils'
+import { Channel, Task, buffers, channel } from 'redux-saga'
+import {
+  all,
+  call,
+  cancel,
+  fork,
+  put,
+  select,
+  takeLatest,
+  take
+} from 'typed-redux-saga'
+
+import { make } from 'common/store/analytics/actions'
+import { prepareStemsForUpload } from 'pages/upload-page/store/utils/stems'
+import * as errorActions from 'store/errors/actions'
+import { reportToSentry } from 'store/errors/reportToSentry'
+import { waitForWrite } from 'utils/sagaHelpers'
+
+import { getUnclaimedPlaylistId } from '../cache/collections/utils/getUnclaimedPlaylistId'
+import { trackNewRemixEvent } from '../cache/tracks/sagas'
+import { retrieveTracks } from '../cache/tracks/utils'
+import { adjustUserField } from '../cache/users/sagas'
+import { addPlaylistsNotInLibrary } from '../playlist-library/sagas'
+
+import { processTrackForUpload, recordGatedTracks } from './sagaHelpers'
+
+const { updateProgress } = uploadActions
+
+type ProgressAction = ReturnType<typeof updateProgress>
+
+const MAX_CONCURRENT_UPLOADS = 4
```

Comment:
I wonder if we should increase this. Your uploads get spread across other nodes these days and we also have a decent number of nodes with available compute. I would vote 6 to 10